src/wpool_pool.erl

```diff
@@ -248,6 +250,16 @@ wpool_size(Name) ->
 -spec next(pos_integer(), wpool()) -> wpool().
 next(Next, WPool) -> WPool#wpool{next = Next}.
 
+-spec add_callback_module(wpool:name(), module()) -> ok | {error, term()}.
+add_callback_module(Pool, Module) ->
+  EventManager = event_manager_name(Pool),
+  ok = gen_event:call(EventManager, wpool_process_callbacks, {add_callback, Module}).
```

Comment:
Why instead of having a single event handler and adding callbacks to its list, don't we let `gen_event` take care of that and just add a new event handler per module?
This function would use `gen_event:add_handler(EventManager, {wpool_process_callbacks, Module}, Module)` and the one below `gen_event:delete_handler(EventManager, {wpool_process_callbacks, Module}).`